modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -71,8 +71,9 @@ export class {{classname}} {
             this.configuration = configuration;
         }
         if (typeof this.configuration.basePath !== 'string') {
-            if (Array.isArray(basePath) && basePath.length > 0) {
-                basePath = basePath[0];
+            const firstBasePath = Array.isArray(basePath) ? basePath[0] : undefined;
+            if (firstBasePath) {
```

Comment:
i would argue we should allow an empty string here as well, i.e.

```suggestion
            if (firstBasePath != null) {
```

Reply:
@macjohnny sorry created new PR https://github.com/OpenAPITools/openapi-generator/pull/18457